src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/service/ReportingApiService.kt

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.digital.hmpps.nomisuserrolesapi.service
+
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import uk.gov.justice.digital.hmpps.nomisuserrolesapi.config.ReportingConfiguration
+import java.nio.charset.Charset
+
+@Service
+@EnableConfigurationProperties(ReportingConfiguration::class)
+class ReportingApiService(
+  @Qualifier("reportingApiWebClient") private val webClient: WebClient,
+  private val config: ReportingConfiguration
+) {
+  fun getReportingUrl(username: String): ReportingUrlResponse =
+    webClient.post()
+      .uri("/reporting/get-url")
+      .bodyValue(
+        ReportingUrlRequest(
+          username = username,
+          clientId = config.client.id,
+          clientSecret = config.client.secret,
+        )
+      )
+      .retrieve()
+      .bodyToMono(ReportingUrlResponse::class.java)
+      .onErrorMap(WebClientResponseException.NotFound::class.java) { it.mapTo(503) }
+      .onErrorMap(WebClientResponseException.InternalServerError::class.java) { it.mapTo(503) }
+      .block()!!
```

Comment:
Can't think of a better way to do this; I want 401/403 to be propagated back, but any other error mapped to 503, so choose the 2 main ones (404/500) but could list every error.
I couldn't used the super class since that converted all errors to 503